src/EditorFeatures/VisualBasic/LineCommit/CommitBufferManager.vb

```diff
@@ -191,9 +207,12 @@ Namespace Microsoft.CodeAnalysis.Editor.VisualBasic.LineCommit
 
                 If startingStatementInfo.MatchingBlockConstruct IsNot Nothing Then
                     ' If we're expanding backwards because of editing an end construct, we don't wan to run
-                    ' expensive semantic formatting checks.  We really just want to fix up indentation.
-                    formattingInfo.UseSemantics = finalSpanStart <= startingStatementInfo.MatchingBlockConstruct.SpanStart
-
+                    ' expensive semantic formatting checks.  We really just want to fix up indentation unless we are in a VB block.
+                    ' In VB some constructs are not "Blocks" in the same way that they are in C#, IsExecutableBlock maps the semantics
+                    ' of VB to match C# Block.
```

Comment:
i don't really get what this comment means.  instead of a vague comment, use exact cases that this counts for to make it clear when this comes up and why we want the specialized bheavior you're adding.  Thanks!

Reply:
I improved the comment BUT the details are really in IsExecutableBlock as they are complex.